### PR TITLE
Separate pipeline dummy data generation from execution/testing

### PR DESCRIPTION
**Pull request by zigaLuksic**
Not all cases are ported ATM as agreed.

### CODE REVIEW

File: eogrow/utils/testing.py [hunk 1]
```diff
@@ -300,6 +300,78 @@ def check_pipeline_logs(pipeline: Pipeline) -> None:
...
+
+    for config in raw_configs:
+        output_folder_key = output_folder_key or config.get("output_folder_key")
+
+        pipeline = load_pipeline_class(config).from_raw_config(config)
+
+        if reset_output_folder:
+            if output_folder_key is None:
+                raise ValueError(
+                    "Pipeline does not have a `output_folder_key` parameter, `output_folder_key` must be set by hand."
```
Comment (mlubej) [inline eogrow/utils/testing.py:326]:
```suggestion
                    "Pipeline does not have an `output_folder_key` parameter, it must be set by hand."
```

File: eogrow/utils/testing.py [hunk 2]
```diff
@@ -300,6 +300,78 @@ def check_pipeline_logs(pipeline: Pipeline) -> None:
...
+    output_folder_key: Optional[str] = None,
+    save_new_stats: bool = False,
+) -> None:
+    """Compares the results from a pipeline run with the saved statistics
+
+    :param config_path: A path to the config file
+    :param stats_path: A path to the file containing result statistics
+    :param output_folder_key: Type of the folder containing results of the pipeline, inferred from config if missing
+    :param save_new_stats: If True then the current results are saved as the statistics. Otherwise, the saved results
+        are compared to the current ones.
```
Comment (mlubej) [inline eogrow/utils/testing.py:349]:
```suggestion
    :param save_new_stats: Save new result stats and skip the comparison. Default is `False`.
```

File: eogrow/utils/testing.py [hunk 3]
```diff
@@ -300,6 +300,78 @@ def check_pipeline_logs(pipeline: Pipeline) -> None:
...
+    config_path: str,
+    *,
+    output_folder_key: Optional[str] = None,
+    reset_output_folder: bool = True,
+) -> None:
+    """Runs a pipeline (or multiple) and checks the logs that all the executions were successful
+
+    :param config_path: A path to the config file
+    :param output_folder_key: Type of the folder containing results of the pipeline, inferred from config if None
+    :param reset_output_folder: If True it will delete content of the folder with results before running the pipeline
```
Comment (mlubej) [inline eogrow/utils/testing.py:313]:
```suggestion
    :param reset_output_folder: Delete the content of the results folder before running the pipeline. Default is `True`.
```

File: eogrow/utils/testing.py [hunk 4]
```diff
@@ -300,6 +300,78 @@ def check_pipeline_logs(pipeline: Pipeline) -> None:
...
+        are compared to the current ones.
+    """
+    crude_configs = collect_configs_from_path(config_path)
+    raw_configs = [interpret_config_from_dict(config) for config in crude_configs]
+
+    for config in raw_configs:
+        output_folder_key = output_folder_key or config.get("output_folder_key")
+        if output_folder_key is None:
+            raise ValueError(
+                "Pipeline does not have a `output_folder_key` parameter, `output_folder_key` must be set by hand."
```
Comment (mlubej) [inline eogrow/utils/testing.py:358]:
```suggestion
                "Pipeline does not have an `output_folder_key` parameter, it must be set by hand."
```

**Review by mlubej — APPROVED** [review]

File: eogrow/utils/testing.py (hunk 3 above)
Comment (zigaLuksic) [reply eogrow/utils/testing.py:313]: defaults are kinda visible :thinking:  if you remove the last sentence i'm ok with it

File: eogrow/utils/testing.py (hunk 2 above)
Comment (zigaLuksic) [reply eogrow/utils/testing.py:349]: same comment about defaults

**Pull request merged**